google-basemap/index.ts

```diff
@@ -0,0 +1,138 @@
+import './style.css';
+import {GoogleMapsOverlay as DeckOverlay} from '@deck.gl/google-maps';
+import {colorCategories} from '@deck.gl/carto';
+import {VectorTileLayer, vectorQuerySource} from '@deck.gl/carto';
+import {Loader} from '@googlemaps/js-api-loader';
+
+const GOOGLE_MAPS_API_KEY = import.meta.env.VITE_GOOGLE_API_KEY;
+const GOOGLE_MAP_ID = import.meta.env.VITE_GOOGLE_MAP_IDS;
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+const accessToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+const connectionName = import.meta.env.VITE_API_CONNECTION_NAME;
+const cartoConfig = {apiBaseUrl, accessToken, connectionName};
+
+let map: google.maps.Map | null = null;
+const overlay = new DeckOverlay({
+  layers: []
+});
+
+const formatNumber = number =>
+  new Intl.NumberFormat('en-US', {
+    notation: 'compact',
+    compactDisplay: 'short'
+  }).format(number);
+
+function setTooltip({x, y, object}) {
+  const tooltip = document.getElementById('tooltip');
+  if (object && tooltip) {
+    tooltip.style.display = 'block';
+    tooltip.style.left = x + 'px';
+    tooltip.style.top = y + 'px';
+    tooltip.innerHTML = `<div style="padding: 8px">Continent: ${object.properties.continent_name}</br>Country: ${object.properties.country_name}</br>Population: ${object.properties.pop_2015}</div>`;
+    console.log(x, y, object);
+  } else if (tooltip) {
+    tooltip.style.display = 'none';
+  }
+}
+
+function render() {
+  const source = vectorQuerySource({
+    ...cartoConfig,
+    sqlQuery: `SELECT geom, country_name, continent_name, pop_2015 FROM cartobq.public_account.world_population_2015 WHERE pop_2015 between @min and @max ORDER BY pop_2015 DESC`,
+    queryParameters: {
+      min: Number(1),
+      max: selectedMaxPopulation
+    }
+  });
+
+  const layers = [
+    new VectorTileLayer({
+      id: 'population',
+      data: source,
+      opacity: 1,
+      pickable: true,
+      getFillColor: colorCategories({
```

Comment:
Could you define this outside the `render` function as `getColorForContinent`? As `render` is tied to a slider `colorCategories` will be needlessly executed many times